docs/testplan/GCU-Dynamic-ACL-testplan.md

```diff
@@ -0,0 +1,421 @@
+
+- Verify that packets that were previously dropped are now forwarded
+
+### Test Case # 4 - Create forward rules within custom ACL table
+
+#### Test Objective
+
+Verify that we can create a forward rule utilizing GCU, and that we can create forward rules for both ipv4 and ipv6
+
+#### Testing Steps
+
+- Use GCU to create 2 new forwarding rules with top priority on our ACL Table, one for IPv4 and one for IPv6
+
+- Use GCU to create drop rule on a specific port with lower priority in ACL Table
+
+- Verify that all operations were successful
+
+- Verify that for both rules created, "show acl rule | grep {rulename}" matches expected output for both rules
+
+- Verify that packets matching forwarding rules on this specific port are correctly forwarded
+
+- Verify that packets not matching forwarding rules on this specific port are correctly dropped
+
+### Test Case # 5 - Replace the IP Address on an ACL Rule
+
+#### Test Objective
+
+Verify that after creation, ACL Rules can have their match conditions updated
+
+#### Testing Steps
+
+- Use GCU to create 2 new forwarding rules on ACL Table
+
+- Use GCU to create drop rule on a specific port with lower priority on ACL Table
+
+- Use GCU to replace the IP addresses in both forwarding rules
+
+- Verify that all operations were successful
+
+- Verify that the results of "show acl rule | grep {rule_name}" matches expected output for both rules
+
+- Verify that packets with IPs matching original forwarding rules on this specific port are dropped
+
+- Verify that packets with IPs matching replacement rules on this specific port are forwarded
+
+### Test Case # 6 - Remove forward rule from ACL Table
+
+#### Test Objective
+
+Verify that after creation, a forward ACL rule can be removed and packets matching the forward rule are no longer forwarded
+
+#### Testing Steps
+
+- Use GCU to create 2 new forwarding rules on ACL Table
+
+- Use GCU to create drop rule on a specific port with lower priority on ACL Table
+
+- Use GCU to remove the 2 forwarding rules
+
+- Verify that all operations were successful
+
+- Verify that the results of "show acl rule {rule_name}" are empty for both rule names
+
+- Verify that packets with IPs matching the removed forwarding rules are dropped on this specific port
+
+### Test Case # 7 - Replace the IP Address of a non-existent ACL Rule
+
+#### Test Objective
+
+Verify that attempting to replace the address of a rule that does not exist properly results in an error and does not affect configDB
+
+#### Testing Steps
+
+- Create 2 new forwarding rules on ACL Table
+
+- Replace the IP addresses in non-existent forwarding rules
+
+- Verify that the replace action failed
+
+### Test Case # 8 - Remove non-existent ACL Table
+
+#### Test Objective
+
+Verify that attempting to remove an ACL Table that does not exist properly results in an error and does not affect configDB
+
+#### Testing Steps
+
+- Attempt to remove a table that does not exist
+
+- Verify that this removal fails
+
```

Comment:
We may need one more case here to verify the device has the capability for configure the required number of ACL rules.

Reply:
I will add a test case to confirm that application of ~100 forward rules and ~20 drop rules works, and that a randomly selected packet that matches either the forward rules/drop ports is correctly forwarded/dropped